### PR TITLE
Display message on failure to get lock on format.json in fs mode on startup

### PR DESCRIPTION
**Pull request by kannappanr**
Retry to see if the lock is free. Retry time will increase binomially.

## Description
Currently, when we run minio which has older version of format.json without the deployment ID like `minio.RELEASE.2018-07-10T01-42-11Z` . Then run the latest version of minio on the same data directory, the executable waits indefinitely to get a lock on `format.json` to update it with the `deployment ID`. This PR fixes that by displaying a message that gives more info to the user and keeps waiting for the user to fix to take action and fix this issue.

## Motivation and Context
Based on a slack conversation. The user over there was using gateway nas, but the issue is essentially the same. He was running three `minio` processes which were sharing the same NFS backend. He was upgrading one of the `minio` processes and was seeing this issue.

## Regression
No

## How Has This Been Tested?
Run `minio.RELEASE.2018-07-10T01-42-11Z`  on FS mode on data directory `/tmp/fs`
Get the latest `minio` release on the same data directory. You can see it hang

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/format-fs.go [hunk 1]
```diff
@@ -333,11 +333,17 @@ func formatFSFixDeploymentID(fsFormatPath string) error {
...
 	for {
 		wlk, err := lock.TryLockedOpenFile(fsFormatPath, os.O_RDWR, 0)
 		if err == lock.ErrAlreadyLocked {
-			// Lock already present, sleep and attempt again.
+			// Lock already present, sleep and attempt again
+			// until for about 5 seconds.
 			time.Sleep(100 * time.Millisecond)
+			attempts++
+			if attempts == 50 {
+				return fmt.Errorf("%s: %s", fsFormatPath, err.Error())
```
Comment (vadmeste) [inline cmd/format-fs.go]: Looks like we only need to show a better error message for this and use your suggestion @kannappanr before approving this PR

**Review by vadmeste — CHANGES_REQUESTED** [review]
One comment

File: cmd/format-fs.go (hunk 1 above)
Comment (kannappanr) [reply cmd/format-fs.go]: done

**Comment by harshavardhana** [comment]
Can this PR be rebased @kannappanr ?

**Comment by kannappanr** [comment]
@harshavardhana Rebased and changed the behavior to display a message and check based on binomially increasing wait time, instead of exiting.

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]
LGTM

**Comment by kannappanr** [comment]
Ran Mint tests on FS setup locally. 
```
docker run -e SERVER_ENDPOINT=192.168.1.83:9000 -e ACCESS_KEY=minio -e SECRET_KEY=minio123 -e ENABLE_HTTPS=0 minio/mint:edge
Running with
SERVER_ENDPOINT:      192.168.1.83:9000
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            core
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 770c77428ce6:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 41 seconds
(4/13) Running aws-sdk-ruby tests ... done in 2 seconds
(5/13) Running awscli tests ... done in 1 minutes and 36 seconds
(6/13) Running mc tests ... done in 5 seconds
(7/13) Running minio-dotnet tests ... done in 8 seconds
(8/13) Running minio-go tests ... done in 0 seconds
(9/13) Running minio-java tests ... done in 2 seconds
(10/13) Running minio-js tests ... done in 33 seconds
(11/13) Running minio-py tests ... done in 13 seconds
(12/13) Running s3cmd tests ... done in 12 seconds
(13/13) Running security tests ... done in 0 seconds

All tests ran successfully

```

**Pull request merged**